eisbuk-admin/src/components/atoms/AttendanceCard/__tests__/AttendanceCard.test.tsx

```diff
@@ -1,28 +1,40 @@
 import React from "react";
 import { render, screen } from "@testing-library/react";
-
 import AttendanceCard from "../AttendanceCard";
 
-// import * as attendanceOperations from "@/store/actions/attendanceOperations";
+import * as attendanceOperations from "@/store/actions/attendanceOperations";
 import { Category } from "eisbuk-shared";
 import { dummySlot } from "../__testData__/dummyData";
 
-/**
- * Mocked `markAttendance` function, expect it to have been called when needed
- */
-// const mockMarkAttendance = jest.spyOn(attendanceOperations, "markAttendance");
+const mockMarkAttImplementation = (customerId: string, attended: boolean) => ({
+  customerId,
+  attended,
+});
+jest
+  .spyOn(attendanceOperations, "markAttendance")
+  .mockImplementation(mockMarkAttImplementation);
 
+const mockDispatch = jest.fn();
+jest.mock("react-redux", () => ({
+  useDispatch: () => mockDispatch,
+}));
 describe("AttendanceCard", () => {
   // write all of the tests (grouped by another describe) within this block
   describe("Smoke test", () => {
     beforeEach(() => {
       render(<AttendanceCard {...dummySlot} />);
     });
     test("should render props", () => {
-      expect(screen.getByText("13:00 - 15:00"));
+      screen.getByText("13:00 - 15:00");
       screen.getByText("Saul");
       screen.getByText("Walter");
       screen.getByText(Category.Competitive);
     });
+    test("should dispatch markAttendance with correct args", () => {
+      screen.getByText("Mark Walter as present").click();
+      expect(mockDispatch).toHaveBeenCalledWith(
+        mockMarkAttImplementation("heisenberg", true)
+      );
+    });
```

Comment:
This isn't a smoke test anymore. This should be under `describe("Test marking attendance functionality", () => {...})`